### PR TITLE
Add test cases

### LINKED ISSUE #26: [Feature] Add some test cases
Add more test cases into project

### PR DESCRIPTION
**Pull request by RazorBach**
Add unit test cases to the project:
Test the data generator and some basic functionalities of each activity.

resolve #26

### CODE REVIEW

**Review by hilaryhuo — APPROVED** [review]
looks good to me!

File: AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/ActionActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    assertNotNull(adapter);
+    AppAction appAction = activity.appAction;
+    assertEquals(adapter.getItemCount(), getCount(appAction));
+  }
+
+  //the number of all actions and fulfillment options
+  private int getCount(AppAction appAction) {
+    int cnt = 0;
+    for(AppActionProtos.Action action : appAction.getActionsList()) {
+      cnt += action.getFulfillmentOptionCount() + 1;
```
Comment (jianjinjian) [inline AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/ActionActivityTest.java:56]: some comment why the extra +1?

File: AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/MainActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class MainActivityTest {
+
+  @Rule
+  public ActivityTestRule<MainActivity> rule = new ActivityTestRule<>(MainActivity.class, false, true);
+  /**
+   * assert that view pager is correctly initialized
```
Comment (jianjinjian) [inline AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/MainActivityTest.java]: nit : Assert and . in the end

File: AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/MainActivityTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+   */
+  @Test
+  public void viewPagerIsCorrect() {
+    MainActivity activity = rule.getActivity();
+    ViewPager viewPager = activity.findViewById(R.id.view_pager);
+    assertNotNull(viewPager);
+    SectionsPagerAdapter adapter = (SectionsPagerAdapter) viewPager.getAdapter();
+    assertNotNull(adapter);
+    //the viewpager adapter should have same size with tab titles
```
Comment (jianjinjian) [inline AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/MainActivityTest.java]: space after // and Start the comment with capitalize 
https://google.github.io/styleguide/javaguide.html#s7-javadoc

**Review by jianjinjian — APPROVED** [review]

File: AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/ActionActivityTest.java (hunk 1 above)
Comment (RazorBach) [reply AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/ActionActivityTest.java:56]: Comments added

File: AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/MainActivityTest.java (hunk 2 above)
Comment (RazorBach) [reply AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/MainActivityTest.java]: Fixed!

File: AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/MainActivityTest.java (hunk 3 above)
Comment (RazorBach) [reply AppActionVisualizer/app/src/androidTest/java/com/example/appactionvisualizer/ui/activity/MainActivityTest.java]: Fixed and also fixed comments in other files.

**Comment by RazorBach** [comment]
All fixed, thanks for reviewing!

**Pull request merged**